### PR TITLE
[UT-1350] Add loading button prop

### PR DESCRIPTION
**Pull request by syntichsizer**

### CODE REVIEW

File: packages/button/src/new-button.tsx [hunk 1]
```diff
@@ -1,5 +1,8 @@
 import { IMTheme as theme } from '@im-ui/theme';
 import styled, { FlattenSimpleInterpolation, css } from 'styled-components';
+import { Spinner } from 'react-activity';
+import 'react-activity/dist/Spinner.css';
```
Comment (dunklesToast) [inline packages/button/src/new-button.tsx:4]: Do we really need to add an additional 7kB dependency? Adding a spinner should be be possible in a better way.

File: packages/button/stories/button.stories.tsx [hunk 2]
```diff
@@ -42,9 +42,14 @@ storiesOf('Button', module)
...
       <>
+        <div>Loading primary:</div>
+        <IMButton buttonType="primary" disabled isLoading>
+          {text('Label', 'Hello World')}
+          <Icon icon="bars" />
+        </IMButton>
         <div>Primary:</div>
         <IMButton buttonType="primary">
-          {text('Label', 'Hello World')}
+          {text('Label', 'Hello world')}
```
Comment (dunklesToast) [inline packages/button/stories/button.stories.tsx:52]: why lowercase it here?

File: packages/button/src/new-button.tsx (hunk 1 above)
Comment (syntichsizer) [reply packages/button/src/new-button.tsx:4]: I guess I could use something like [this](https://loading.io/css/).

But we already have the same spinner [here](https://github.com/Instamotion/im-components/pull/326), so I'd have to update StyledDropdown components for them to look the same :/

File: packages/button/stories/button.stories.tsx (hunk 2 above)
Comment (syntichsizer) [reply packages/button/stories/button.stories.tsx:52]: No idea why I did this..

File: packages/button/src/new-button.tsx [hunk 3]
```diff
@@ -65,7 +69,15 @@ const renderDisabledButton = (
     `) as FlattenSimpleInterpolation;
 };
 
-export const Button = styled.button<ButtonProps>`
+const ButtonComponent = styled.button<React.HTMLProps<HTMLButtonElement> & ButtonProps>`
   ${props => renderBaseButton(props.buttonType || 'primary', props?.width, props?.size)}
   ${props => renderDisabledButton(props.buttonType || 'primary', props.disabled)}
-`;
+` as React.ComponentType<ButtonProps>;
```
Comment (dhamid-im) [inline packages/button/src/new-button.tsx:75]: why is `as React.ComponentType<ButtonProps>` needed?

File: packages/button/src/new-button.tsx [hunk 4]
```diff
@@ -65,7 +69,15 @@ const renderDisabledButton = (
...
+
+export const Button = (props: React.HTMLProps<HTMLButtonElement> & ButtonProps) => {
```
Comment (dhamid-im) [inline packages/button/src/new-button.tsx:77]: i see that `React.HTMLProps<HTMLButtonElement> & ButtonProps` is used twice .. why not create a `type` declaration to reuse it?

File: packages/button/src/new-button.tsx (hunk 1 above)
Comment (dunklesToast) [reply packages/button/src/new-button.tsx:4]: hmm. so the first approach would add another spinner to the button, right?
The second approach would make Dropdown a dependency of Button? That would not be great

Comment (syntichsizer) [reply packages/button/src/new-button.tsx:4]: Not really, I'll try something else :))

Comment (dhamid-im) [reply packages/button/src/new-button.tsx:4]: also just to add to it, why are we importing it from the "dist" folder? don't we have a css loader configured to move files automatically?

**Review by jogamod — APPROVED** [review]

**Review by AntonTretiak — APPROVED** [review]

**Review by dhamid-im — APPROVED** [review]

**Pull request merged**